### PR TITLE
Table indexing config validation

### PR DESCRIPTION
**Pull request by icefury71**
Related to #5942 

## Description
This is to ensure any column name mentioned in the table indexing config is valid (i.e. it exists in the corresponding schema). Also checking for space in the table name.

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
No

Does this PR fix a zero-downtime upgrade introduced earlier?
No

Does this PR otherwise need attention when creating release notes?
No

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java [hunk 1]
```diff
@@ -225,4 +229,54 @@ private static void validateTierConfigList(@Nullable List<TierConfig> tierConfig
       }
     }
   }
+
+  private static void validateIndexingConfig(@Nullable IndexingConfig indexingConfig, @Nullable Schema schema) {
+    if (indexingConfig == null || schema == null) {
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java:234]: Can you also add FieldConfig validation here?

File: pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java [hunk 2]
```diff
@@ -65,6 +68,7 @@ public static void validate(TableConfig tableConfig, @Nullable Schema schema) {
     validateValidationConfig(tableConfig, schema);
     validateIngestionConfig(tableConfig.getIngestionConfig(), schema);
     validateTierConfigList(tableConfig.getTierConfigsList());
+    validateIndexingConfig(tableConfig.getIndexingConfig(), schema);
```
Comment (jackjlli) [inline pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java:71]: Could you update the description of the method `validate` as well?

File: pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java [hunk 3]
```diff
@@ -225,4 +229,54 @@ private static void validateTierConfigList(@Nullable List<TierConfig> tierConfig
...
+      for (String columnName : indexingConfig.getSortedColumn()) {
+        columnNameToConfigMap.put(columnName, "Sorted Column Config");
+      }
+    }
+    if (indexingConfig.getVarLengthDictionaryColumns() != null) {
+      for (String columnName : indexingConfig.getVarLengthDictionaryColumns()) {
+        columnNameToConfigMap.put(columnName, "Var Length Column Config");
+      }
+    }
+
```
Comment (npawar) [inline pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java:274]: can we include star tree index columns here?

Comment (icefury71) [reply pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java:307]: added

File: pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java (hunk 2 above)
Comment (icefury71) [reply pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java:74]: Done

File: pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java (hunk 1 above)
Comment (icefury71) [reply pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java:242]: Done

**Review by siddharthteotia — APPROVED** [review]
LGTM

File: pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java [hunk 4]
```diff
@@ -225,4 +233,99 @@ private static void validateTierConfigList(@Nullable List<TierConfig> tierConfig
...
+    List<StarTreeIndexConfig> starTreeIndexConfigList = indexingConfig.getStarTreeIndexConfigs();
+    if (starTreeIndexConfigList != null) {
+      for (StarTreeIndexConfig starTreeIndexConfig : starTreeIndexConfigList) {
+        // Dimension split order cannot be null
+        for (String columnName : starTreeIndexConfig.getDimensionsSplitOrder()) {
+          columnNameToConfigMap.put(columnName, "StarTreeIndex Config");
+        }
+        // Function column pairs cannot be null
+        for (String functionColumnPair : starTreeIndexConfig.getFunctionColumnPairs()) {
+          String[] functionColumnArray = functionColumnPair.split("__");
```
Comment (npawar) [inline pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java:291]: can we use the `AggregationFunctionColumnPair#fromColumnName` or reuse the delimiter field in this class?

**Comment by npawar** [comment]
As discussed offline, it would be nice to have some validations on index columns which cannot be in noDictionary columns (e.g. if column has inverted index, it should not be in noDictionary).
Are you planning to add that as part of same PR?

**Comment by icefury71** [comment]
> As discussed offline, it would be nice to have some validations on index columns which cannot be in noDictionary columns (e.g. if column has inverted index, it should not be in noDictionary).
> Are you planning to add that as part of same PR?

Not in this PR. I'll create a separate one for that.

File: pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java (hunk 4 above)
Comment (icefury71) [reply pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java]: Done

**Pull request merged**